fees/fees.go

```diff
@@ -0,0 +1,59 @@
+package fees
+
+import (
+	"github.com/btcsuite/btcd/btcjson"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcutil"
+)
+
+// GetDetailsFunc is a function which looks up transactions by hash.
+type GetDetailsFunc func(hash *chainhash.Hash) (*btcjson.TxRawResult, error)
+
+// CalculateFee returns the total fees for the transaction provided.
+// TODO(carla): identify change address and split fees between outputs.
+func CalculateFee(details GetDetailsFunc, txid *chainhash.Hash) (btcutil.Amount,
+	error) {
+
+	var fees btcutil.Amount
+
+	tx, err := details(txid)
+	if err != nil {
+		return 0, err
+	}
+
+	// First, we minus total outputs from our fees.
```

Comment:
not a big deal, but seems more natural to add inputs first, the subtract outputs?